test/core/ref_func.wast

```diff
@@ -74,10 +74,37 @@
   "unknown function 7"
 )
 
-(assert_invalid
-  (module (func $f) (global funcref (ref.func $f)))
-  "undeclared function reference"
+
+;; Reference declaration
+
+(module
+  (func $f1)
+  (func $f2)
+  (func $f3)
+  (func $f4)
+  (func $f5)
+  (func $f6)
+
+  (table $t 1 funcref)
+
+  (global funcref (ref.func $f1))
+  (export "f" (func $f2))
+  (elem (table $t) (i32.const 0) func $f3)
+  (elem (table $t) (i32.const 0) funcref (ref.func $f4))
+  (elem func $f5)
+  (elem funcref (ref.func $f6))
```

Comment:
Should the function referenced by the start section be included here?

Reply:
Excellent question! I think not, since the start function doesn't escape. I added another special case and test for it.

Reply:
That's true, but neither do non-exported globals and tables. It does seem like a strange omission.

Reply:
@binji, I suppose you mean the references stored in such globals or tables? Those can still escape after `global.get` or `table.get` and returning the result to somewhere. I think checking that these can't escape would require inter-procedural flow analysis.

Reply:
Oh right, good point.